astropy/utils/tests/test_data.py

```diff
@@ -1006,7 +1005,9 @@ def test_check_download_cache(tmpdir, temp_cache, valid_urls, invalid_urls):
     testurl2, testurl2_contents = next(valid_urls)
 
     zip_file_name = os.path.join(tmpdir, "the.zip")
-    clear_download_cache()
+    with pytest.warns(CacheMissingWarning, match='Not clearing data cache - '
```

Comment:
@aarchiba , I made it much more explicit now. So if indeed it is just my machine acting weird, the jobs would fail in CI.

Reply:
This is still likely to be trouble - it will fail on any machine (like mine) that doesn't emit the warning. There's a filterwarnings context manager that would succeed in ignoring the warning but I think it's better to prevent the warning being emitted - it is fine if clear_download_cache() is called when no cache exists, and we shouldn't be emitting warnings.

Wait. I think what's going on is maybe that the shelve is opened, then the on-disk file is deleted, then the shelve is closed, and this backend doesn't like that. Which is not unreasonable. But I can catch that exception and let it go without a warning, because this only happens with a no-argument clear_download_cache().